lightning/src/ln/onion_utils.rs

```diff
@@ -1374,30 +1376,29 @@ where
 		res = Some(FailureLearnings {
 			network_update,
 			short_channel_id,
-			payment_failed_permanently: error_code & PERM == PERM && is_from_final_non_blinded_node,
+			payment_failed_permanently: error_code.is_permanent() && is_from_final_non_blinded_node,
 			failed_within_blinded_path: false,
 		});
 
-		let (description, title) = errors::get_onion_error_description(error_code);
 		if debug_field_size > 0 && err_packet.failuremsg.len() >= 4 + debug_field_size {
 			log_info!(
 				logger,
-				"Onion Error[from {}: {}({:#x}) {}({})] {}",
+				"Onion Error[from {}: {:?}({:#x}) {}({})] {}",
 				route_hop.pubkey(),
-				title,
 				error_code,
+				error_code.failure_code(),
 				debug_field,
 				log_bytes!(&err_packet.failuremsg[4..4 + debug_field_size]),
-				description
+				error_code
```

Comment:
nit: no need to log `error_code` twice, similar below

Reply:
I thought it would be nice to log the variant name (`{:?}`) and the string (`{}`) so that the variant was easy to look up.

Will update based on [whether we keep display](https://github.com/lightningdevkit/rust-lightning/pull/3744/commits/29f0ccb085c5c200cf68de12285995a1bf10077f#r2068586298).